### PR TITLE
[quant][core][gpu][bug-fix] Added additional caching support in quantized cudnn add_relu op

### PR DESCRIPTION
**Pull request by dzdang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #75772

Summary:
Previous caching strategy for quantized cudnn add_relu operator was insufficient
as it did not properly record all the necesary information. This PR adds several
items to the CacheKey (e.g., input sizes, input dimensions, etc..) that enables
proper caching

Test plan:
```
python test/test_quantization.py -k test_qadd_relu_cudnn
```

Differential Revision: [D35634278](https://our.internmc.facebook.com/intern/diff/D35634278)

### CODE REVIEW

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35634278).

File: aten/src/ATen/native/quantized/cudnn/BinaryOps.cpp [hunk 1]
```diff
@@ -21,16 +22,43 @@
...
   uint8_t output_alignment;
   bool kReluFused;
 };
-std::unordered_map<CacheKey, cudnn_frontend::ManagedOpaqueDescriptor, at::native::ParamsHash<CacheKey>, at::native::ParamsEqual<CacheKey>> execution_plan_cache;
+void setAddParams(
+    AddParams* params, const at::Tensor& input_a, const at::Tensor& input_b,
+    bool deterministic, bool allow_tf32) {
+  // operator datatype needs to be int32 for int8 matmul, but we can
+  // set the datatype for output tensor to int32 or fp32
+  memset(params, 0, sizeof(AddParams));
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/cudnn/BinaryOps.cpp:47]: do we need to do this?

File: aten/src/ATen/native/quantized/cudnn/BinaryOps.cpp [hunk 2]
```diff
@@ -21,16 +22,43 @@
...
   uint8_t input_b_alignment;
   uint8_t output_alignment;
   bool kReluFused;
 };
-std::unordered_map<CacheKey, cudnn_frontend::ManagedOpaqueDescriptor, at::native::ParamsHash<CacheKey>, at::native::ParamsEqual<CacheKey>> execution_plan_cache;
+void setAddParams(
+    AddParams* params, const at::Tensor& input_a, const at::Tensor& input_b,
+    bool deterministic, bool allow_tf32) {
+  // operator datatype needs to be int32 for int8 matmul, but we can
+  // set the datatype for output tensor to int32 or fp32
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/cudnn/BinaryOps.cpp:46]: looks like this comment is not related?

**Review by jerryzh168 — APPROVED** [review]

File: aten/src/ATen/native/quantized/cudnn/BinaryOps.cpp (hunk 1 above)
Comment (dzdang) [reply aten/src/ATen/native/quantized/cudnn/BinaryOps.cpp:47]: I copied that over from https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cudnn/ConvShared.cpp#L89. 
I think it's used for initialization and to set the remaining values of `input_*_size` to zero. Without it, I think we'd just need an additional loop to initialize the remaining values.

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35634278).
